runtime-api/bonded-currency/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use parity_scale_codec::{alloc::string::String, Codec};
+use sp_std::vec::Vec;
+
+mod pool_details;
+pub use pool_details::*;
+
+sp_api::decl_runtime_apis! {
+	/// Runtime API to compute the collateral for a given amount and pool ID
+	/// and to query all pool IDs where the given account is the manager or owner.
+	#[api_version(1)]
+	pub trait BondedCurrency<Balance, PoolId, Operation, AccountId, BitType, PoolDetails, Error> where
+		Balance: Codec,
+		PoolId: Codec,
+		Operation: Codec,
+		AccountId: Codec,
+		BitType: Codec,
+		PoolDetails: Codec,
+		Error: Codec
+		{
+			/// Calculates the collateral for the given amount.
+			/// The operation is determining whether the amount is minted or burned.
+			/// The calculated collateral amount is based on the current state of the pool.
+			fn get_quota(
+				pool_id: PoolId,
+				currency_idx: u8,
+				operation: Operation,
```

Comment:
I think the `Operation` type should not be generic. While it makes sense for the other ones, in this case it will be either `Burn` or `Mint`, with an associated value, so I would probably collapse this and the `amount` into a single argument, that is not generic.

Reply:
I think it makes sense to keep Operation generic. If we want to introduce another possible operation such as Swap, we can easily update the trait implementation. But yes combining amount with the operation is a good suggestion.

Reply:
I am referring to adding a new operation and first test that in the test network

Reply:
So basically only allowing 2 operations on production but 3 in development? Ok that makes sense, but than you need to remove the definition from the runtime API crate itself, and make it implement some sort of `get_amount() -> Balance` if you want to keep it generic.

Reply:
[e12f907](https://github.com/KILTprotocol/kilt-node/pull/828/commits/e12f907916bfe519fc2bce7f7150574431a59c7f)

Reply:
I dropped the trait. I found it a bit unnecessary. Let me know what you think.